test/async-hooks/test-zctx.js

```diff
@@ -0,0 +1,31 @@
+const common = require('../common')
+const assert = require('assert')
+const initHooks = require('./init-hooks')
+const { checkInvocations } = require('./hook-checks')
+const zlib = require('zlib')
+
+const hooks = initHooks()
+
+hooks.enable()
+zlib.deflate('..........', common.mustCall(ondeflate));
+
+function ondeflate() {
+  const as = hooks.activities()
+  const a = as[0]
+  checkInvocations(a, { init: 1, before: 2, after: 2 }, 'while in ondeflate callback')
```

Comment:
I don't really know why either, there was just limited time to do this that morning. We should definitely investigate, it may be a bug in the async-hooks impl.